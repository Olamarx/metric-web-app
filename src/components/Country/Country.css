.wholeClass {
  height: 100vh;
  background-color: pink;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.navStyle {
  margin-left: 150px;
  margin-top: 50px;
  font-size: 20px;
}

.date {
  font-size: 15px;
  margin-top: 50px;
}

.twoImg {
  margin-top: 20px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.twoImg img {
  width: 200px;
  height: 200px;
}

.countryInd {
  width: 300px;
  height: 300px;
}

.countryCoat {
  width: 150px;
  height: 150px;
}

.wholeStat {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.wholeStat ul {
  list-style: none;
  display: flex;
  width: 100%;
  flex-direction: column;
  gap: 10px;
}

.indLI {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-between;
}

.no-coatOfArm {
  width: 100px;
  height: 100px;
  font-size: 12px;
  margin-top: 20px;
}

@media screen and (min-width: 992px) {
  .date {
    font-size: 30px;
    margin-top: 50px;
  }

  .wholeStat ul {
    list-style: none;
    display: flex;
    width: 100%;
    flex-direction: column;
    gap: 50px;
  }

  .no-coatOfArm {
    width: 300px;
    height: 300px;
    font-size: 20px;
    margin-top: 20px;
  }

  .countryInd {
    width: 500px;
    height: 500px;
  }

  .countryCoat {
    width: 300px;
    height: 300px;
  }

  .navStyle {
    margin-left: 150px;
    margin-top: 50px;
    font-size: 40px;
  }
}
