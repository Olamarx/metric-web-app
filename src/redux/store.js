import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './Countries';
// import countryDetails, { loadCountry } from './MoreDetails';

const store = configureStore({
  reducer: {
    countriesReducer,
  },
});

export default store;
