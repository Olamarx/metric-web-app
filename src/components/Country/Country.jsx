import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';
import { setCountry } from '../../redux/country/country';
import { getpollution, removePollution } from '../../redux/currentPollution/currentPollution';
import Damages from '../damages/Damages';
import Spinner from '../Loading/Loading';
import './Country.css';

function Country() {
  const dispatch = useDispatch();
  const continent = useSelector((state) => state.continent);
  const country = useSelector((state) => state.country);
  const pollution = useSelector((state) => state.pollution);

  const countrySelect = (country) => {
    setCountry(dispatch, country);
  };

  const revert = () => {
    countrySelect({});
    removePollution(dispatch);
  };

  const stat = {
    1: ['Good', '#0f0'],
    2: ['Fair', '#66ff00'],
    3: ['Moderate', '#eeff00'],
    4: ['Poor', '#ff7300'],
    5: ['Very Poor', '#ff0000'],
  };
  useEffect(() => {
    dispatch(getpollution);
    dispatch(removePollution);
  }, []);

  return (
    <div className="wholeClass">
      <div className="navStyle">
        <NavLink
          to="/continents"
          onClick={revert}
        >
          <span>
            <FontAwesomeIcon icon={faArrowLeft} />
            {continent.toUpperCase()}
          </span>
        </NavLink>
        <span>{` <<<<< ${country.name}`}</span>
      </div>
      <div
        className="date"
      >
        <span>
          As at
          {` ${new Date(pollution.dt * 1000).toLocaleString().replace(/[/]/gi, '-')}
        `}
        </span>
        <span>{` ${country.name}'s Population is ${country.population}`}</span>
        <div className="twoImg">
          <img src={country.flag} alt={country.name} className="countryInd" />
          { country.coatOfArm ? (<img src={country.coatOfArm} alt={country.coatOfArm} className="countryCoat" />)
            : (
              <div className="no-coatOfArm">
                {`The Coat of Arm for ${country.name}
                is yet to be added, please check back later.`}
              </div>
            )}
        </div>
        {pollution && Object.keys(pollution).length > 0
          ? (
            <div className="wholeStat">
              <span
                style={{ color: stat[pollution.index][1] }}
              >
                {`${pollution.index}: ${stat[pollution.index][0]}`}
              </span>
              <ul>
                {
                        Object.keys(pollution.composition).map((key) => {
                          const com = key.replace('_', '.');
                          const comName = com.replace(/[0-9]/g, '').toUpperCase();
                          const comNum = com.replace(/[a-z]/g, '');
                          return (
                            <li key={key} className="indLI">
                              <span>
                                {comName}
                                <sub>{comNum}</sub>
                              </span>

                              <span>
                                {pollution.composition[key]}
                                <small>
                                  &#181;g/m
                                  <sup>3</sup>
                                </small>
                              </span>
                            </li>
                          );
                        })
                    }
              </ul>
            </div>
          ) : (<Spinner />)}
      </div>
      {pollution && Object.keys(pollution).length > 0
        ? <Damages inten={parseInt(pollution.index, 10)} col={stat[pollution.index][1]} /> : <></>}
    </div>
  );
}

export default Country;
