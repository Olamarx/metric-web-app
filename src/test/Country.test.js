import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Country from '../components/Country/Country';
import store from '../redux/store';

describe('The indivial page of the app countires', () => {
  it('renders properly', async () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Country />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
