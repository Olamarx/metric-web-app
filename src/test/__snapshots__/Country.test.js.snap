// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The indivial page of the app countires renders properly 1`] = `
<div
  className="wholeClass"
>
  <div
    className="navStyle"
  >
    <a
      className=""
      href="/continents"
      onClick={[Function]}
    >
      <span>
        AFRICA
      </span>
    </a>
    <span>
       &lt;&lt;&lt;&lt;&lt; 
    </span>
  </div>
  <div
    className="date"
  >
    <span>
      As at
       Invalid Date
        
    </span>
    <span>
       's Population is undefined
    </span>
    <div
      className="twoImg"
    >
      <img
        alt=""
        className="countryInd"
      />
      <div
        className="no-coatOfArm"
      >
        The Coat of Arm for 
                is yet to be added, please check back later.
      </div>
    </div>
    <div
      className="ui active dimmer"
    >
      <div
        className="ui big text loader"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
